### ISSUE TITLE
Tab completion for company mode doesn't work in evil mode

### ISSUE THREAD

**Issue by abrahamparayil** [emacs-evil/evil-collection#499]
Hey,
First of all thanks for `evil-collection`!

### Expected Behavior
Without evil mode, when I start typing company starts showing me suggestions and I can move the selection to the option I want and press `<tab>` to insert the completion.

I was hoping for something similar here. Basically I want company to insert a selection when I press `<tab>`.

### Actual Behaviour
Now when I press `<tab>` it just cycles through the options without ever inserting the selection.

### Config
- company:
```emacs-lisp
(use-package company :ensure t
  :bind (:map company-active-map ("<tab>" . company-complete-selection)))
(add-hook 'after-init-hook 'global-company-mode)
```
- I am using the configuration from the readme of evil-collection. I also have `evil-leader` installed.

**Comment by condy0919** [comment]
https://github.com/company-mode/company-mode/blob/master/company-tng.el#L153

**Comment by abrahamparayil** [comment]
@condy0919 I updated my config to:
```emacs-lisp
(use-package evil-collection
  :after evil
  :ensure t
  :config
  (evil-collection-init)
  (setq evil-collection-use-company-tng nil))
```
It still does not complete when I hit `<tab>`.

**Comment by condy0919** [comment]
Set before loads or use `:custom`.

An example of mine: https://github.com/condy0919/.emacs.d/blob/master/lisp/init-evil.el#L44-L61

**Comment by abrahamparayil** [comment]
Updated the config but no change:
```emacs-lisp
(use-package evil-collection
  :after evil
  :ensure t
  :custom (setq evil-collection-use-company-tng nil)
  :config
  (evil-collection-init))
```

**Comment by condy0919** [comment]
No need of setq use in `:custom`

**Comment by abrahamparayil** [comment]
Updated, no change. When the suggestions appear I press `<tab>` but it just goes to the next suggestion and doesn't insert the suggestion but doing `M-x company-complete-selection` does work.

**Comment by condy0919** [comment]
Weired. Can you change the value of `evil-collection-company-use-tng` in evil-collection?

The file is located at `~/.emacs.d/elpa/evil-collection/` directory.

**Comment by jojojames** [comment]
Can you provide an emacs -Q recipe instead?

**Comment by abrahamparayil** [comment]
> Weird. Can you change the value of evil-collection-company-use-tng in evil-collection?

@condy0919 I tried this and still no change.

**Comment by condy0919** [comment]
> > Weird. Can you change the value of evil-collection-company-use-tng in evil-collection?
> 
> @condy0919 I tried this and still no change.

Would you mind to provide a Emacs -Q recipe please? Or I can't guess what's wrong.

**Comment by abrahamparayil** [comment]
@condy0919 https://pastebin.com/raw/UfwfdWbD

Would this be enough? It has both my company and evil stuff.

**Comment by jojojames** [comment]
I'd think you'd want a more minimal version than that but firstly,

we'd just want to figure out if evil-collection is somehow blocking 'normal' tab functionality and not that you can't do your special TAB completion behavior (that'd be out of scope). Of course, if evil-collection is somehow stopping you from setting up TAB to do what you want, that'd be relevant too. 

Can you remove this line:   :bind (:map company-active-map ("<tab>" . company-complete-selection))) and let us know what the behavior looks like?

Also, when I've bound TAB, I bound multiple tabs:

e.g.

    (let ((keymap company-active-map))
      (define-key keymap [return] 'j-company-expand-yasnippet-or-return)
      (define-key keymap (kbd "RET") 'j-company-expand-yasnippet-or-return)
      (define-key keymap "\C-n" 'company-select-next)
      (define-key keymap [tab] 'company-select-next)
      (define-key keymap (kbd "TAB") 'company-select-next)
      (define-key keymap [S-tab] 'company-select-previous)
      (define-key keymap [S-iso-lefttab] 'company-select-previous)
      (define-key keymap [(shift tab)] 'company-select-previous)
      (define-key keymap [backtab] 'company-select-previous)
      (define-key keymap "\C-p" 'company-select-previous)
      (define-key keymap [backtab] 'company-select-previous)
      (define-key keymap (kbd "S-TAB") 'company-select-previous)
      (define-key keymap (kbd "DEL") company-cancel-or-backspace)
      (define-key keymap [backspace] company-cancel-or-backspace)
      (define-key keymap (kbd "<backspace>") company-cancel-or-backspace))

**Comment by jojojames** [comment]
> @jojojames I think we should set it to nil by default. Right?

Yeah, I think it's not worth my time answering this question anymore. Lets set up a PR to -remove- the flag and any relevant configuration from evil-collection around that value. We can tag a stable release before pushing that PR and that should be the end of that.

**Comment by abrahamparayil** [comment]
> Can you remove this line: :bind (:map company-active-map ("" . company-complete-selection))) and let us know what the behavior looks like?
I did this, even then pressing tab just goes to the next suggestion instead of making the actual completion.

**Comment by condy0919** [comment]
It should be fixed in evil-collection-20210707.1929
